core-client/src/main/java/org/glassfish/jersey/client/authentication/BasicAuthenticator.java

```diff
@@ -96,11 +99,33 @@ public boolean filterResponseAndAuthenticate(ClientRequestContext request, Clien
                     .getCredentials(request, defaultCredentials, HttpAuthenticationFilter.Type.BASIC);
 
             if (credentials == null) {
+                if (response.hasEntity()) {
+                    discardInputAndClose(response.getEntityStream());
+                }
                 throw new ResponseAuthenticationException(null, LocalizationMessages.AUTHENTICATION_CREDENTIALS_MISSING_BASIC());
             }
 
             return HttpAuthenticationFilter.repeatRequest(request, response, calculateAuthentication(credentials));
         }
         return false;
     }
+
```

Comment:
This discardInputAndClose is once again in HttpAuthenticationFilter in PR3860.  Can you put it to the AuthenticationUtil as discussed in PR 3862?